packages/icons/src/icon/index.js

```diff
@@ -3,6 +3,20 @@
  */
 import { cloneElement } from '@wordpress/element';
 
+// Disable reason: JSDoc linter doesn't seem to parse the union (`&`) correctly.
+/* eslint-disable jsdoc/valid-types */
+/** @typedef {{icon: JSX.Element} & {size?: number} & import('react').ComponentPropsWithoutRef<'SVG'>} IconProps */
```

Comment:
Can we merge the two statically known props?

```suggestion
/** @typedef {{icon: JSX.Element, size?: number} & import('react').ComponentPropsWithoutRef<'SVG'>} IconProps */
```

Reply:
Yes! Good catch, thanks @aduth!